### DISCUSSION TITLE
How to merge two jsondocument?

### DISCUSSION THREAD

**Discussion by SKYC829** [dotnet/aspnetcore#55618] [Q&A]
I have a json named a:

``` json
{
"propA":"myProp",
"some_other_prop":1234
}
```

and a json named b:

``` json
{
"propA":"myPropPlus",
"propB": true
}
```

how can i merge these two json by using System.Text.Json.JsonDocument ?
the expected result:

``` json
{
"propA":"myPropPlus",
"propB": true,
"some_other_prop":1234
}
```

**Comment by markmouton22** [comment]
Hi :)

**An example:**

`using System;
using System.Text.Json;
using System.Collections.Generic;
					
public class Program
{
	public static void Main()
	{
		string json1 = @"{
    ""propA"":""myProp"",
    ""some_other_prop"":1234
}";

string json2 = @"{
    ""propA"":""myPropPlus"",
    ""propB"": true
}";

JsonDocument doc1 = JsonDocument.Parse(json1);
JsonDocument doc2 = JsonDocument.Parse(json2);

var dict1 = new Dictionary<string, JsonElement>();
foreach (var element in doc1.RootElement.EnumerateObject())
{
    dict1[element.Name] = element.Value;
}

var dict2 = new Dictionary<string, JsonElement>();
foreach (var element in doc2.RootElement.EnumerateObject())
{
    dict2[element.Name] = element.Value;
}

foreach (var item in dict2)
{
    dict1[item.Key] = item.Value;
}

string mergedJson = JsonSerializer.Serialize(dict1);
Console.WriteLine(mergedJson);
	}
}`

**Result:**

{"propA":"myPropPlus","some_other_prop":1234,"propB":true}
